src/InOneWeekend/camera.h

```diff
@@ -56,54 +56,90 @@ class camera {
     }
 
   private:
-    int image_height;
-    point3 origin;
-    vec3 horizontal;
-    vec3 vertical;
-    point3 lower_left_corner;
-    vec3 u, v, w;
-    double lens_radius;
+    int    image_height;    // Rendered image height
+    point3 origin;          // Camera ray origin
+    point3 pixel00_loc;     // Location of pixel 0, 0
+    vec3   pixel_delta_u;   // Offset to pixel to the right
+    vec3   pixel_delta_v;   // Offset to pixel below
+    vec3   viewport_u;      // Vector across viewport horizontal edge
+    vec3   viewport_v;      // Vector down viewport vertical edge
+    vec3   u, v, w;         // Camera frame basis vectors
+    vec3   defocus_disk_u;  // Defocus disk horizontal radius
+    vec3   defocus_disk_v;  // Defocus disk vertical radius
 
     void initialize() {
+        image_width  = (image_width < 1) ? 1 : image_width;
```

Comment:
This feels unnecessary?

Reply:
If I understand correctly that this is to stop the person from set the width to 0 or lower, but that feels like the kind of thing that the user should just not do? And if they accidentally do it, they should be able to debug

Reply:
We must do this for `image_height`, since a width of 1 may yield a height of 0 for some aspect ratios. I put this here for symmetry, but you're right that it's strictly unnecessary given our current style of "here's a gun, there's your foot, have a great day".

I'll pull this line.